internal/log_analysis/log_processor/parsers/awslogs/alb.go

```diff
@@ -178,3 +178,34 @@ func (event *ALB) updatePantherFields(p *ALBParser) {
 	event.AppendAnyDomainNamePtrs(event.DomainName)
 	event.AppendAnyAWSARNPtrs(event.ChosenCertARN, event.TargetGroupARN)
 }
+
+func (p *ALBParser) extractRequestParams(record []string) (map[string]string, error) {
```

Comment:
We should pass the csv fields directly to keep csv index handling in the parent function.

```suggestion
func (p *ALBParser) extractRequestParams(requestLine, classificationReason string) (map[string]string, error) {
```
We can then call this with `p.extractRequestParameters(record[12], record[28])`.
This will make unit testing the method easier also.
Reading `record[12]` or `record[28]` in the middle of this function seems like something that should be someone else's concern.

Reply:
We could but this is somehow debatable. It seems to leak extraction details to the main method, whereas it could be entirely the function's concern to retrieve the request parameters from the record. The helper field extraction method is also not public, but rather wraps some of the `Parse` method logic, so it is expected to be tied to internal field position knowledge.

As far as unit testing is concerned, since the method is private, I don't think we can/should test it, but rather perform more extensive testing through `Parse`, if necessary.

Finally, if you notice, the classification reason is not utilized anywhere else (not sure if we should add it at some point). 

Thoughts?

Reply:
My view on this is the inverse. Having the arbitrary `28` index inside the 'inner' function seems like passing irrelevant information to a function that should be a simple mapping of RequestLing -> url parameters.
The function needs not be a method of `p` either.
It is a 'pure' function.
It should not even depend on the `SpaceInUri` thing TBH. 

A request line is valid as long as we can split it into `{method} {uri} {version}`. In this regard, it could just be
```go
func splitRequestLine(line string) (method, uri, version string)
```
The caller can handle the special cases having more context than the 'split' function.


All this is a discussion of code structure. The actual PR will work fine as is. I am just trying to convey my preferences regarding code aesthetics and composability here.

Reply:
On a micro level the current `parseRequestParameters`
has to allocate a `[]string` for the parts *and* a `map[string]sting` _on each line_.
ALB events are quite chatty. Having a zero-allocation split here might be noticeable in terms of performance.

```golang
func splitRequestLine(line string) (method, uri, version string) {
   if pos := strings.IndexByte(line, ' '); 0 <= pos && pos < len(line) {
       method, line := line[:pos], line[pos+1:]
       if pos := strings.LastIndexByte(line, ' '); 0 <= pos && pos < len(line) {
         uri, version := line[:pos], line[pos+1:]
         return method, uri, version
       }
   }
   return line, "", ""
}
```

Again, this is all just educational. I am not requesting this to be changed.

Reply:
> The function needs not be a method of p either.

True, I'll change that.

Your observations make sense as more generic considerations, but I believe you are thinking it in terms of a usable and intuitive public API, which is not the case here though. This is actually functionality that is tied to `Parse`, but just extracted to a separate function for clarity (and so that we don't end up with a 300-line function at some point 😄 ). Thus it makes sense to have knowledge of these esoteric parameters (field position, classification reason hint). If you visualize the whole function body inlined in `Parse` then I believe you'll likely agree that it makes more sense to be written like this.

Reply:
> Having a zero-allocation split here might be noticeable in terms of performance.

Good note, I'll research it a bit more, thanks.

Reply:
Function simplified significantly in 6393ee5a0fbccabeacb90f8b3c07d46ffce0f5b3.